File: SysGestionTurismo/src/main/java/pe/edu/upeu/sysgestionturismo/repositorio/UsuarioRepository.java
package pe.edu.upeu.sysgestionturismo.repositorio;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import pe.edu.upeu.sysgestionturismo.modelo.Usuario;

import java.util.Optional;

public interface UsuarioRepository extends JpaRepository<Usuario, Long> {

    @Query("SELECT u FROM Usuario u WHERE u.correo = :correo")
    Optional<Usuario> findByCorreo(@Param("correo") String correo);
}

File: SysTurismoJPC/app/src/main/java/pe/edu/upeu/systurismojpc/repository/UsuarioRepository.kt
package pe.edu.upeu.systurismojpc.repository

import pe.edu.upeu.systurismojpc.modelo.LoginRequest
import pe.edu.upeu.systurismojpc.modelo.LoginResponse
import retrofit2.Response

interface UsuarioRepository {
    suspend fun loginUsuario(loginRequest: LoginRequest): Response<LoginResponse>
}
